lib/sprite.dart

```diff
@@ -4,18 +4,22 @@ import 'extensions/offset.dart';
 import 'extensions/vector2.dart';
 import 'palette.dart';
 
+class SpriteOpts {
```

Comment:
Is this break out really necessary when there are only two values stored on here?
The other one makes sense since it has so many.

Reply:
I don't have strong opinions on that, I decided to use a new class to create a "standard" and be somehow similar to the SpriteAnimation, but both would work for me.

Reply:
I think I still prefer keeping few arguments out of "opts"-classes, it's easy to end up in that pattern everywhere otherwise and too much of it would probably be considered an anti-pattern.

Reply:
I think I prefer keeping this to Animation as well, Sprite is sufficiently simple and there is only one way to construct it (see my comment about animationOpt)

Reply:
opt classes are useful for:
- when options can live by itself and have inter-instance relations like `TextStyle` from flutter has `.apply` and `.compareTo` etc
- when we don't want to impact diferent places with API changes, ex:  adding or removing options here would not require to refactor on this `Sprite`contructor and `loadSprite` in the previous file. 
- users could extend it and change behavior, like `Decoration` form flutter

Reply:
Since the arguments here are optional I don't see how those points are valid?